### ISSUE TITLE
Autocompletion fails in julia-mode

### ISSUE THREAD

**Issue by whilo** [syl20bnr/spacemacs#11822]
#### Description :octocat:

Autocompletion is not working in `julia-mode` buffers of the julia-layer. @non-Jedi
Using Julia `1.0.3`

#### Reproduction guide :beetle:
- Start Emacs
- Edit a julia buffer
- Observe that there are no completions (on hitting TAB)

*Observed behaviour:* :eyes: :broken_heart:
The autocomplete layer does not show any completions in `julia-mode`.

Autocomplete is somewhat working in Julia REPL.

*Expected behaviour:* :heart: :smile:
The autocomplete layer works as expected for Julia (including user defined) functions in `julia-mode`. 

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 2c28c7d21)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(fsharp javascript html helm auto-completion better-defaults emacs-lisp git markdown
        (org :variables org-enable-reveal-js-support t)
        (shell :variables shell-default-height 30 shell-default-position 'bottom)
        shell-scripts spell-checking syntax-checking version-control clojure python latex bibtex go ess julia display)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11

**Comment by gdkrmr** [comment]
There are currently several breaking changes. `LanguageServer.jl` just released a julia 1.0 compatible version which is incompatible with the current `lsp-julia` and `lsp-mode` on MELPA introduced breaking changes. How to make it work see #11723.

**Comment by gdkrmr** [comment]
This should work now if you install `LanguageServer.jl` correctly @whilo.
